// @vuepress/markdown/lib/plugins/codePlugin/languages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.languageYaml = exports.languageTypescript = exports.languageStylus = exports.languageRust = exports.languageRuby = exports.languagePython = exports.languageMarkdown = exports.languageKotlin = exports.languageJavascript = exports.languageFsharp = exports.languageDocker = exports.languageCsharp = exports.languageBash = void 0;
exports.languageBash = {
    name: 'bash',
    ext: 'sh',
    aliases: ['bash', 'sh', 'shell', 'zsh'],
};
exports.languageCsharp = {
    name: 'csharp',
    ext: 'cs',
    aliases: ['cs', 'csharp'],
};
exports.languageDocker = {
    name: 'docker',
    ext: 'docker',
    aliases: ['docker', 'dockerfile'],
};
exports.languageFsharp = {
    name: 'fsharp',
    ext: 'fs',
    aliases: ['fs', 'fsharp'],
};
exports.languageJavascript = {
    name: 'javascript',
    ext: 'js',
    aliases: ['javascript', 'js'],
};
exports.languageKotlin = {
    name: 'kotlin',
    ext: 'kt',
    aliases: ['kotlin', 'kt'],
};
exports.languageMarkdown = {
    name: 'markdown',
    ext: 'md',
    aliases: ['markdown', 'md'],
};
exports.languagePython = {
    name: 'python',
    ext: 'py',
    aliases: ['py', 'python'],
};
exports.languageRuby = {
    name: 'ruby',
    ext: 'rb',
    aliases: ['rb', 'ruby'],
};
exports.languageRust = {
    name: 'rust',
    ext: 'rs',
    aliases: ['rs', 'rust'],
};
exports.languageStylus = {
    name: 'stylus',
    ext: 'styl',
    aliases: ['styl', 'stylus'],
};
exports.languageTypescript = {
    name: 'typescript',
    ext: 'ts',
    aliases: ['ts', 'typescript'],
};
exports.languageYaml = {
    name: 'yaml',
    ext: 'yml',
    aliases: ['yaml', 'yml'],
};
