// @mr-hope/vuepress-plugin-pwa/lib/client/PWAInstall.js
import Vue from "vue";
import PWAInstallModal from "./PWAInstallModal.vue";
import { locales } from "./define";
export default Vue.extend({
    name: "PWAInstall",
    components: { PWAInstallModal },
    data: () => ({
        canInstall: false,
        hasRelatedApps: false,
        isOpen: false,
        isIOS: false,
        isSafari: false,
        hinted: false,
    }),
    computed: {
        install() {
            return locales[this.$localePath || "/"].install;
        },
        useHint() {
            return this.isIOS && this.isSafari && this.hinted === false;
        },
        showInstall() {
            return (this.hasRelatedApps && this.canInstall) || this.useHint;
        },
    },
    mounted() {
        if (this.getInstalledStatus()) {
            const { userAgent } = navigator;
            // handle iOS specifically
            this.isIOS =
                // regular iPhone
                userAgent.includes("iPhone") ||
                    // regular iPad
                    userAgent.includes("iPad") ||
                    // iPad pro
                    Boolean(userAgent.includes("Macintosh") &&
                        navigator.maxTouchPoints &&
                        navigator.maxTouchPoints > 2);
            this.isSafari =
                navigator.userAgent.includes("Safari") && !userAgent.includes("Chrome");
            this.hinted = Boolean(localStorage.getItem("iOS-pwa-hint"));
        }
        if ("getInstalledRelatedApps" in navigator)
            void navigator
                .getInstalledRelatedApps()
                .then((result) => {
                this.hasRelatedApps = result.length > 0;
            });
    },
    methods: {
        getInstalledStatus() {
            if (navigator.standalone)
                return navigator.standalone;
            return matchMedia("(display-mode: standalone)").matches;
        },
        hint() {
            this.isOpen = false;
            this.hinted = true;
            // do not notify again
            localStorage.setItem("iOS-pwa-hint", "hinted");
        },
    },
});
