// @vuepress/utils/lib/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logger = exports.createError = exports.error = exports.warn = exports.success = exports.tip = exports.info = void 0;
const chalk = require("chalk");
const info = (...args) => {
    console.log(chalk.cyan('info'), ...args);
};
exports.info = info;
const tip = (...args) => {
    console.log(chalk.blue('tip'), ...args);
};
exports.tip = tip;
const success = (...args) => {
    console.log(chalk.green('success'), ...args);
};
exports.success = success;
const warn = (...args) => {
    console.warn(chalk.yellow('warning'), ...args);
};
exports.warn = warn;
const error = (...args) => {
    console.error(chalk.red('error'), ...args);
};
exports.error = error;
const createError = (message) => {
    (0, exports.error)(message);
    return new Error(message);
};
exports.createError = createError;
exports.logger = {
    info: exports.info,
    tip: exports.tip,
    success: exports.success,
    warn: exports.warn,
    error: exports.error,
    createError: exports.createError,
};
