// @vuepress/client/dist/index.js
import {
  ClientOnly,
  Content,
  Vuepress,
  pageData,
  pageDataEmpty,
  pageFrontmatterSymbol,
  pageHeadSymbol,
  pageHeadTitleSymbol,
  pageLangSymbol,
  pagesData,
  resolvers,
  routeLocaleSymbol,
  siteData,
  siteLocaleDataSymbol,
  updateHeadSymbol,
  usePageData,
  usePageFrontmatter,
  usePageHead,
  usePageHeadTitle,
  usePageLang,
  usePagesData,
  useRouteLocale,
  useSiteData,
  useSiteLocaleData,
  useUpdateHead,
  withBase
} from "./chunk-RTLFYRTT.js";

// src/helpers/defineClientConfig.ts
var defineClientConfig = (clientConfig) => clientConfig;
export {
  ClientOnly,
  Content,
  Vuepress,
  defineClientConfig,
  pageData,
  pageDataEmpty,
  pageFrontmatterSymbol,
  pageHeadSymbol,
  pageHeadTitleSymbol,
  pageLangSymbol,
  pagesData,
  resolvers,
  routeLocaleSymbol,
  siteData,
  siteLocaleDataSymbol,
  updateHeadSymbol,
  usePageData,
  usePageFrontmatter,
  usePageHead,
  usePageHeadTitle,
  usePageLang,
  usePagesData,
  useRouteLocale,
  useSiteData,
  useSiteLocaleData,
  useUpdateHead,
  withBase
};
