// @vuepress/core/lib/pluginApi/createPluginApiHooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createPluginApiHooks = void 0;
const createHookQueue_1 = require("./createHookQueue");
const createPluginApiHooks = () => ({
    // life cycle hooks
    onInitialized: (0, createHookQueue_1.createHookQueue)('onInitialized'),
    onPrepared: (0, createHookQueue_1.createHookQueue)('onPrepared'),
    onWatched: (0, createHookQueue_1.createHookQueue)('onWatched'),
    onGenerated: (0, createHookQueue_1.createHookQueue)('onGenerated'),
    // extends hooks
    extendsMarkdownOptions: (0, createHookQueue_1.createHookQueue)('extendsMarkdownOptions'),
    extendsMarkdown: (0, createHookQueue_1.createHookQueue)('extendsMarkdown'),
    extendsPageOptions: (0, createHookQueue_1.createHookQueue)('extendsPageOptions'),
    extendsPage: (0, createHookQueue_1.createHookQueue)('extendsPage'),
    extendsBundlerOptions: (0, createHookQueue_1.createHookQueue)('extendsBundlerOptions'),
    // client config file hooks
    clientConfigFile: (0, createHookQueue_1.createHookQueue)('clientConfigFile'),
    // bundler hooks
    alias: (0, createHookQueue_1.createHookQueue)('alias'),
    define: (0, createHookQueue_1.createHookQueue)('define'),
});
exports.createPluginApiHooks = createPluginApiHooks;
