// @vuepress/markdown/lib/markdown.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMarkdown = void 0;
const MarkdownIt = require("markdown-it");
const plugins_1 = require("./plugins");
const utils_1 = require("./utils");
/**
 * Create vuepress customized markdown-it instance
 */
const createMarkdown = ({ anchor, assets, code, customComponent, emoji, extractHeaders, extractTitle, hoistTags, importCode, links, toc, ...markdownItOptions } = {}) => {
    // create raw markdown-it instance
    const md = MarkdownIt({
        ...markdownItOptions,
        // should always enable html option
        html: true,
    });
    // =====================================================
    // following plugins push rules to the end of chain, so
    // the order to use them is important
    // parse emoji
    if (emoji !== false) {
        md.use(plugins_1.emojiPlugin, emoji);
    }
    // add anchor to headers
    if (anchor !== false) {
        md.use(plugins_1.anchorPlugin, {
            level: [1, 2, 3, 4, 5, 6],
            slugify: utils_1.slugify,
            permalink: plugins_1.anchorPlugin.permalink.ariaHidden({
                class: 'header-anchor',
                symbol: '#',
                space: true,
                placement: 'before',
            }),
            ...anchor,
        });
    }
    // allow toc syntax
    if (toc !== false) {
        md.use(plugins_1.tocPlugin, {
            level: [2, 3],
            slugify: utils_1.slugify,
            linkTag: 'RouterLink',
            ...toc,
        });
    }
    // extract headers into env
    if (extractHeaders !== false) {
        md.use(plugins_1.extractHeadersPlugin, {
            level: [2, 3],
            slugify: utils_1.slugify,
            ...extractHeaders,
        });
    }
    // extract title into env
    if (extractTitle !== false) {
        md.use(plugins_1.extractTitlePlugin);
    }
    // =====================================================
    // following plugins modify or replace the rule in place
    // and have no conflicts, so the order is not important
    // treat unknown html tags as custom components
    if (customComponent !== false) {
        md.use(plugins_1.customComponentPlugin);
    }
    // replace relative link of assets with absolute link
    if (assets !== false) {
        md.use(plugins_1.assetsPlugin, assets);
    }
    // hoist vue SFC blocks and extract them into env
    if (hoistTags !== false) {
        md.use(plugins_1.hoistTagsPlugin, hoistTags);
    }
    // process external and internal links
    if (links !== false) {
        md.use(plugins_1.linksPlugin, links);
    }
    // process code fence
    if (code !== false) {
        md.use(plugins_1.codePlugin, code);
    }
    // handle import_code syntax
    if (importCode !== false) {
        md.use(plugins_1.importCodePlugin, importCode);
    }
    return md;
};
exports.createMarkdown = createMarkdown;
